sample/src/main/java/io/ona/kujaku/sample/activities/MainActivity.java

```diff
@@ -253,14 +334,42 @@ public void onReceive(Context context, Intent intent) {
                                 String resultStatus = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_STATUS);
                                 MapboxOfflineDownloaderService.SERVICE_ACTION serviceAction = (MapboxOfflineDownloaderService.SERVICE_ACTION) bundle.get(MapboxOfflineDownloaderService.KEY_RESULTS_PARENT_ACTION);
 
-                                if (resultStatus.equals(MapboxOfflineDownloaderService.SERVICE_ACTION_RESULT.FAILED.name())) {
-                                    String message = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_MESSAGE);
+                                String message = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_MESSAGE);
 
+                                if (MapboxOfflineDownloaderService.SERVICE_ACTION_RESULT.FAILED.name().equals(resultStatus)) {
                                     if (!TextUtils.isEmpty(message)) {
                                         if (!message.contains("MapBox Tile Count limit exceeded")) {
                                             showInfoNotification("Error occurred " + mapUniqueName + ":" + serviceAction.name(), message);
                                         }
                                     }
+
+                                    if (serviceAction == MapboxOfflineDownloaderService.SERVICE_ACTION.DELETE_MAP && !TextUtils.isEmpty(message)) {
+                                        Toasty.error(MainActivity.this, message)
+                                                .show();
+                                    } else if (!TextUtils.isEmpty(mapUniqueName) && mapUniqueName.equals(currentMapDownload)) {
```

Comment:
It's not clear what this conditional tests.

Why would `mapUniqueName` be empty and why would it not be equal to the `currentMapDownload`?

Reply:
Should the `currentMapDownload` be nulled out in case of an error?

Reply:
> Should the currentMapDownload be nulled out in case of an error?

So, there are **network errors** when downloading the map, such errors are sometimes not permanent and do not specifically stop the download from happening

When the internet connection is back and download continues, the button is re-enabled.

Do you have a way go about this or a recommendation about this for the sample app?

Reply:
Sorry, 🙏 it's a convoluted piece of code that's been simplified for the number of cases expected to handle

Reply:
This makes sense, please add some code comments to help explain some of this logic, especially the conditional.